packages/cli/src/sso/saml/saml.service.ee.ts

```diff
@@ -249,25 +250,29 @@ export class SamlService {
 	}
 
 	async fetchMetadataFromUrl(): Promise<string | undefined> {
+		if (!this._samlPreferences.metadataUrl)
+			throw new BadRequestError('Error fetching SAML Metadata, no Metadata URL set');
 		try {
 			// TODO:SAML: this will not work once axios is upgraded to > 1.2.0 (see checkServerIdentity)
 			const agent = new https.Agent({
-				rejectUnauthorized: !this.ignoreSSL,
+				rejectUnauthorized: !this._samlPreferences.ignoreSSL,
 			});
-			const response = await axios.get(this.metadataUrl, { httpsAgent: agent });
+			const response = await axios.get(this._samlPreferences.metadataUrl, { httpsAgent: agent });
 			if (response.status === 200 && response.data) {
 				const xml = (await response.data) as string;
 				const validationResult = await validateMetadata(xml);
 				if (!validationResult) {
 					throw new BadRequestError(
-						`Data received from ${this.metadataUrl} is not valid SAML metadata.`,
+						`Data received from ${this._samlPreferences.metadataUrl} is not valid SAML metadata.`,
 					);
 				}
 				return xml;
 			}
 		} catch (error) {
-			// eslint-disable-next-line @typescript-eslint/restrict-template-expressions
-			throw new BadRequestError(`Error fetching SAML Metadata from ${this.metadataUrl}: ${error}`);
+			throw new BadRequestError(
+				// eslint-disable-next-line @typescript-eslint/restrict-template-expressions
+				`Error fetching SAML Metadata from ${this._samlPreferences.metadataUrl}: ${error}`,
```

Comment:
Do we need the whole error here? Can't we use only the message?

Reply:
The error may contain more context which could be useful.